Avoid email links in Fediverse mentions (edited)

nicodh opened the pull request:
Linkify renders fediverse links like @user@mastodon.social as 
```
<a href="https://mastodon.social/@user">@user@mastodon.social</a>
```

This is also a test how much effort it is to adapt the linkify library

#skip-changelog as this is still experimental and based on https://github.com/deltachat/deltachat-desktop/pull/5502

#public-preview to enable testing

nicodh committed:
Parse and render fediverse links

r10s commented:
yip, i am also sceptical about that, for reason kinked above, we may want to use a similar format ourselves one day, and this PR would make things even harder.

but also wrt consistency  with android/ios

nicodh commented:
> we may want to use a similar format ourselves one day, and this PR would make things even harder.

I think once we want mentions the first step would be to add a select for contacts if a @ sign is entered. The implementation would probably then be displayed as @username but needs to be @contactId or something similar "under the hood". So it will never be a simple @username - especially since we allow spaces etc. in usernames...

nicodh committed:
Parse fediverse links as text

Since the correct use of these links would need
a webfinger request to detect the real target and
after discussing pro & cons we decided not to
render the links for now

nicodh committed:
Fix comment

nicodh committed:
Remove initialText

Remove moved folders

WofWca reviewed (commented):
I don't think we need this at all, looking at the usefulness to complexity ratio.

I did not verify the correctness of the implementation. But the fact that `clearFediverseMentions` takes the whole message and outputs the whole message means that the function could potentially mess it up in unrelated ways.

The unrelated changes look OK (but I'd prefer them to be a part of a separate MR).

I'll need time to understand the code.
